src/screens/atomic-swap/NewSwap.js

```diff
@@ -6,47 +6,43 @@
  */
 import BackButton from "../../components/BackButton";
 import React, { useEffect, useState } from "react";
-import { useHistory } from 'react-router-dom';
 import { t } from "ttag";
 import Loading from "../../components/Loading";
-import { generateEmptyProposalFromPassword, updatePersistentStorage } from "../../utils/atomicSwap";
+import {
+    generateEmptyProposal,
+} from "../../utils/atomicSwap";
 import { useDispatch, useSelector } from "react-redux";
-import { proposalGenerated } from "../../actions";
+import { proposalCreateRequested } from "../../actions";
 
 export default function NewSwap (props) {
-    const [proposalId, setProposalId] = useState('');
     const [password, setPassword] = useState('');
     const [isLoading, setIsLoading] = useState(false);
-    const allProposals = useSelector(state => state.proposals);
     const wallet = useSelector(state => state.wallet);
-    const history = useHistory();
+
+    // Global interactions
+    const lastFailedRequest = useSelector(state => state.lastFailedRequest);
     const dispatch = useDispatch();
 
     const [errorMessage, setErrorMessage] = useState('');
 
-    const navigateToProposal = (pId) => {
-        history.replace(`/wallet/atomic_swap/proposal/${pId}`);
-    }
-
     const createClickHandler = () => {
         if (password.length < 3) {
-            setErrorMessage('Please insert a password more than 3 characters long');
+            setErrorMessage(t`Please insert a password more than 3 characters long`);
             return;
         }
 
         setIsLoading(true);
-        const { data, id } = generateEmptyProposalFromPassword(password, wallet);
-        setProposalId(id);
-        dispatch(proposalGenerated(id, password, data))
+        const newPartialTx = generateEmptyProposal(wallet);
+        dispatch(proposalCreateRequested(newPartialTx, password));
     }
 
     useEffect(() => {
-        // If this proposal exists, navigate to it immediately
-        if (allProposals[proposalId]) {
-            updatePersistentStorage(allProposals);
-            navigateToProposal(proposalId);
+        // Shows the error message if it happens
+        if (lastFailedRequest && lastFailedRequest.message) {
```

Comment:
Looking only at this screen, it would make sense to do so. But when looking at [the saga that is called](https://github.com/HathorNetwork/hathor-wallet/blob/7c750e74d7e2922b56f7d00f8e9fee4ff92c3e15/src/sagas/atomicSwap.js#L128) on click, it either:
- Throws and fills this `lastFailedRequest.message` field or;
- Navigates to the `EditSwap` screen

So, if there is no automatic navigation then we're sure to have an error message. Not having one _and_ staying in this screen would be a strange behavior anyway, and I think it wouldn't require a "This should never happen" kind of branch here, as it would also need to be unit tested later.